src/huggingface_hub/file_download.py

```diff
@@ -496,52 +491,71 @@ def http_get(
         displayed_name = f"(…){displayed_name[-40:]}"
 
     # Stream file to buffer
-    progress = tqdm(
+    with tqdm(
         unit="B",
         unit_scale=True,
         total=total,
         initial=resume_size,
         desc=displayed_name,
         disable=bool(logger.getEffectiveLevel() == logging.NOTSET),
-    )
-    try:
+    ) as progress:
+        if hf_transfer and total is not None and total > 10 * DOWNLOAD_CHUNK_SIZE:
+            supports_callback = "callback" in inspect.signature(hf_transfer.download).parameters
```

Comment:
Might be useful to warn users that a new version of `hf_transfer` supporting progress bars exists. Otherwise, I found it a bit counter-intuitive since a progress bar is displayed but stays at 0% -at if the download is blocked for some reason-.

```suggestion
            supports_callback = "callback" in inspect.signature(hf_transfer.download).parameters
            if not supports_callback:
                warnings.warn("You are using an outdated version of `hf_transfer`. Consider upgrading to latest version to enable progress bars using `pip install -U hf_transfer`.")
```

Reply:
Done https://github.com/huggingface/huggingface_hub/pull/1792/commits/3e4a6033a837309877136596b1bd433563d8e20c